### ISSUE TITLE
Block User (Channel) needs to block on both sides

### ISSUE THREAD

**Issue by JoshWho** [WWBN/AVideo#6954]
When a user blocks another user, the block should also happen on the other channel as well. (Reciprocally) 

Right now it is just a one way block and the person that does the block still sees' the channel and videos.

**Comment by DanielnetoDotCom** [comment]
I am not sure if that is how it should work, but I will think about it.

**Comment by JoshWho** [comment]
It makes sense. If a user is being bullied or harassed it only makes sense to keep them separated.  Block on both sides will achieve that and they can't harass each other anymore.

**Comment by para2022** [comment]
My block system is screwed if i block someone i cannot unblock them and it doesn't actually block them i can still see all their content and the block button doesn't change to unblock, well it does for a split second then changes back to block again, and if i click again on the block button it says they are already blocked lol

**Comment by para2022** [comment]
This user was blocked on his channel and as you can see their is the block button again and i can see his channel and get the popup saying he is blocked if i click the block button again.

![Screenshot 2022-08-06 at 20 17 27](https://user-images.githubusercontent.com/107706817/183263218-2e462a39-79cd-42d1-9f36-6824752160c3.png)

**Comment by JoshWho** [comment]
See if the the other channel had the reciprocal block it wouldn't show the channel anymore.

**Comment by para2022** [comment]
There is nothing at all recorded in the avideo.log file about this issue

**Comment by para2022** [comment]
i get this now:

 [02-Sep-2022 00:10:55 Europe/London] AVideoLog::ERROR: {"type":1,"message":"Uncaught Error: Class 'Cache' not found in \/home\/myMySQLUser\/public_html\/myMySQLUser\/plugin\/ReportVideo\/Objects\/videos_reported.php:154\nStack trace:\n#0 \/home\/myMySQLUser\/public_html\/myMySQLUser\/plugin\/ReportVideo\/ReportVideo.php(243): VideosReported->delete()\n#1 \/home\/myMySQLUser\/public_html\/myMySQLUser\/plugin\/ReportVideo\/block.json.php(38): ReportVideo->unBlock()\n#2 {main}\n  thrown","file":"\/home\/myMySQLUser\/public_html\/myMySQLUser\/plugin\/ReportVideo\/Objects\/videos_reported.php","line":154} SCRIPT_NAME: /plugin/ReportVideo/block.json.php

**Comment by para2022** [comment]
It is this causing the problem in plugin/reportvideo/objects/videos_reported.php:

Cache::deleteFirstPageCache();

on line 145 and 154. Once those lines of code are removed referencing firstpagecache the block works fine with no error. It would also be nice for a popup saying they were blocked like the popup when you report someone. 

Thank you

**Comment by DanielnetoDotCom** [comment]
@para2022 this update should fix it

**Comment by para2022** [comment]
Sorry to say the but system is still broken. If i block someone now it works fine. But i can’t block anymore than once. If i block a second user the bug comes back where they cannot be blocked.

So if i block one person it works perfect but block another and it becomes broken again. I have to clear out the database table [videos_reported] when it breaks for it to actually work again.

Thanks

**Comment by DanielnetoDotCom** [comment]
@para2022  Please as long as your issue has nothing to do with this thread, please open a new issue and place your log file there

**Comment by para2022** [comment]
Will do daniel thanks
